tools/ts-generator/types/test.ts

```diff
@@ -221,4 +222,34 @@ describe("Buffer functions test", function () {
       asserts.strictEqual(iterable.readInt32LE(), 1343530892);
     }
   );
+
+  it('should write and read an Optional on buffer', function () {
+      const io = new IOBuf();
+      const rpcHeader: RpcHeader = {
+          version: 1,
+          compression: 0,
+          correlationId: 0,
+          headerChecksum: 0,
+          meta: 0,
+          payloadChecksum: BigInt(2),
+          payloadSize: 2
+      };
+      const writtenBytes = BF.writeOptional(io, rpcHeader, (item, auxBuffer, auxOffset) => BF.writeObject(auxBuffer, RpcHeader, item));
+      const writtenBytesUndefined = BF.writeOptional(io, undefined, (item, auxBuffer, auxOffset) => BF.writeObject(auxBuffer, RpcHeader, item));
+      const ioIterable = io.getIterable();
```

Comment:
serialize your types into an IOBuf without having to wrap it in an RPC header. This adds a layer of complexity that isn't necessary in order to know if the methods are working or not.

Reply:
Make senses, done